packages/notifier/src/storesub.js

```diff
@@ -67,6 +74,50 @@ export const makeStoredSubscriber = (subscriber, storageNode, marshaller) => {
   return storesub;
 };
 
+export const TopicMetaShape = M.splitRecord(
+  {
+    topic: SubscriberShape,
+    storagePath: M.promise(),
+  },
+  { description: M.string() },
+);
+/**
+ * @template {object} T topic value
+ * @typedef {{
+ *   description?: string,
+ *   topic: ERef<Subscriber<T>>,
+ *   storagePath: ERef<string>,
+ * }} TopicMeta
+ */
+
+export const TopicMetasRecordShape = M.recordOf(M.string(), TopicMetaShape);
+
+/**
+ * @typedef {{
+ *   [topicName: string]: TopicMeta<unknown>,
+ * }} TopicMetasRecord
+ */
+
+/**
+ * @param {Subscriber<any>} topic
+ * @param {ERef<StorageNode>} storageNode
+ * @param {ERef<ReturnType<typeof makeMarshal>>} marshaller
+ * @returns {void}
+ */
+export const pipeTopicToStorage = (topic, storageNode, marshaller) => {
+  assertAllDefined({ topic, storageNode, marshaller });
+
+  const marshallToStorage = makeMarshallToStorage(storageNode, marshaller);
+
+  // Start publishing the source.
+  forEachPublicationRecord(topic, marshallToStorage).catch(err => {
+    // TODO: How should we handle and/or surface this failure?
+    // https://github.com/Agoric/agoric-sdk/pull/5766#discussion_r922498088
+    console.error('followAndStoreTopic failed to iterate', err);
+  });
+};
+harden(pipeTopicToStorage);
+
```

Comment:
Would you put all of this in a file called something like `stored-topic.js`?  I'm trying to distinguish the new Topic APIs from the rest of the backward-compat things in notifier.